src/clinical/service-worker-thread/tasks.ts

```diff
@@ -175,18 +182,19 @@ const mapEntityDocuments = (
   const samples = originalResultsArray.find(result => result.entityName === 'sample_registration');
 
   if (completionRecords.completionStats && samples !== undefined) {
-    const normalSpecimens = samples.results.filter(
-      sample => sample.tumour_normal_designation === 'Normal',
-    );
-    const tumourSpecimens = samples.results.filter(
-      sample => sample.tumour_normal_designation === 'Tumour',
-    );
-    normalSpecimens.forEach(specimen =>
-      updateCompletionTumourStats(specimen, 'normal', completionRecords),
-    );
-    tumourSpecimens.forEach(specimen =>
-      updateCompletionTumourStats(specimen, 'tumour', completionRecords),
+    const sampleResults = samples.results.filter(
+      sample => typeof sample.sample_type === 'string' && !sample.sample_type?.includes('RNA'),
     );
+
+    sampleResults.forEach(sample => {
+      if (
+        sample.tumour_normal_designation &&
+        typeof sample.tumour_normal_designation === 'string'
```

Comment:
`if(typeof sample.tumour_normal_designation === 'string')` should just work as typeof will return undefined for undefined

Reply:
Nice thank you!